homeassistant/components/daikin/config_flow.py

```diff
@@ -22,24 +23,43 @@ class FlowHandler(config_entries.ConfigFlow):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
-    async def _create_entry(self, host, mac):
+    async def _create_entry(self, host, mac, key=None, uuid=None, password=None):
         """Register new entry."""
         # Check if mac already is registered
         for entry in self._async_current_entries():
             if entry.data[KEY_MAC] == mac:
                 return self.async_abort(reason="already_configured")
 
-        return self.async_create_entry(title=host, data={CONF_HOST: host, KEY_MAC: mac})
+        return self.async_create_entry(
+            title=host,
+            data={
+                CONF_HOST: host,
+                KEY_MAC: mac,
+                CONF_KEY: key,
+                CONF_UUID: uuid,
+                CONF_PASSWORD: password,
+            },
+        )
 
-    async def _create_device(self, host):
+    async def _create_device(self, host, key=None, password=None):
         """Create device."""
 
+        # BRP07Cxx devices needs uuid together with key
+        if key is not None and not "":
```

Comment:
The second check is not correct. It will always be true.

Reply:
🤦